bundles/org.openhab.binding.qolsysiq/src/main/java/org/openhab/binding/qolsysiq/internal/handler/QolsysIQZoneHandler.java

```diff
@@ -0,0 +1,118 @@
+/**
+ * Copyright (c) 2010-2022 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.qolsysiq.internal.handler;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.qolsysiq.internal.QolsysIQBindingConstants;
+import org.openhab.binding.qolsysiq.internal.client.dto.event.ZoneActiveEvent;
+import org.openhab.binding.qolsysiq.internal.client.dto.event.ZoneUpdateEvent;
+import org.openhab.binding.qolsysiq.internal.client.dto.model.Zone;
+import org.openhab.binding.qolsysiq.internal.client.dto.model.ZoneStatus;
+import org.openhab.binding.qolsysiq.internal.config.QolsysIQZoneConfiguration;
+import org.openhab.core.library.types.DecimalType;
+import org.openhab.core.library.types.OpenClosedType;
+import org.openhab.core.library.types.StringType;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.thing.binding.BridgeHandler;
+import org.openhab.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link QolsysIQZoneHandler} manages security zones.
+ *
+ * @author Dan Cunningham - Initial contribution
+ */
+@NonNullByDefault
+public class QolsysIQZoneHandler extends BaseThingHandler {
+    private final Logger logger = LoggerFactory.getLogger(QolsysIQZoneHandler.class);
+
+    private int zoneId;
+
+    public QolsysIQZoneHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void initialize() {
```

Comment:
A last comment: you are not overriding `bridgeStatusChanged`. That means, when the partition handler status will changed form OFFLINE to ONLINE, your thing status will changed to ONLINE even if the zone id is wrong for example.
The best would be to override `bridgeStatusChanged` and move most of the code there in it. In `initialize`, you will call `bridgeStatusChanged`. Keep in `initialize` the check about the null bridge.